src/lib/y2storage/disk.rb

```diff
@@ -110,14 +110,11 @@ def preferred_ptable_type
       PartitionTables::Type.find(:gpt)
     end
 
-    def name_or_partition?(name)
-      return true if self.name == name
-
-      partitions.any? { |part| part.name == name }
-    end
-
-    def self.find_by_name_or_partition(devicegraph, name)
-      all(devicegraph).detect { |disk| disk.name_or_partition?(name) }
+    # @see Device#is?
+    #
+    # In this case, true if type is or contains :disk
+    def is?(types)
+      super || types_include?(types, :disk)
```

Comment:
to be honest I do not like much implementation, as it is a lot of repetition and duplication ( in fact only interesting part of this method is `:disk` rest is just sauce. what about defining method types? which return all types of device? like
```
# base
protected def types
  [:device]
end

# mountable
protected def types
  super << :mountable
end

# and snapshot
protected def types
  super << :snapshot
end
```

if you do not mind some magic, it can be done also with constants
```
#base
TYPE = :device

#mountable
TYPE = :mountable

# snapshot
TYPE = :snapshot

# and implementation
def types
  self.class.ancestors.each_with_object([]) do |klass, result|
     klass.const_get("TYPE") if klass.const_defined?("TYPE")
  end
end
```

but in the end it is just style detail, not problem of API itself.